libcaf_core/test/config_option.cpp

```diff
@@ -0,0 +1,238 @@
+/******************************************************************************
+ *                       ____    _    _____                                   *
+ *                      / ___|  / \  |  ___|    C++                           *
+ *                     | |     / _ \ | |_       Actor                         *
+ *                     | |___ / ___ \|  _|      Framework                     *
+ *                      \____/_/   \_|_|                                      *
+ *                                                                            *
+ * Copyright (C) 2011 - 2016                                                  *
+ * Dominik Charousset <dominik.charousset (at) haw-hamburg.de>                *
+ *                                                                            *
+ * Distributed under the terms and conditions of the BSD 3-Clause License or  *
+ * (at your option) under the terms and conditions of the Boost Software      *
+ * License 1.0. See accompanying files LICENSE and LICENSE_ALTERNATIVE.       *
+ *                                                                            *
+ * If you did not receive a copy of the license files, see                    *
+ * http://opensource.org/licenses/BSD-3-Clause and                            *
+ * http://www.boost.org/LICENSE_1_0.txt.                                      *
+ ******************************************************************************/
+
+#include "caf/config.hpp"
+
+#define CAF_SUITE config_option
+#include "caf/test/unit_test.hpp"
+
+#include "caf/config_option.hpp"
+#include "caf/actor_system_config.hpp"
+
+using namespace std;
+using namespace caf;
+
+
+namespace {
+
+struct fixture {
+};
+
+constexpr const char* category = "category";
+constexpr const char* name = "name";
+constexpr const char* explanation = "explanation";
+constexpr size_t line = 0;
+
+template<class T>
+constexpr T init_value() { return 0; }
+
+template<class T, class U, class V> 
+constexpr T failure_value() {
+  return static_cast<V>(std::numeric_limits<U>::max()) + 2;
```

Comment:
Why +2 (as opposed to +1)? `failure_value` is not a good description of the purpose. `overflow`?